lib/magpie_web/router.ex

```diff
@@ -1,8 +1,14 @@
 defmodule Magpie.Router do
   use MagpieWeb, :router
   use Plug.ErrorHandler
+  import Plug.BasicAuth
 
   pipeline :browser do
+    if Application.get_env(:magpie, :no_basic_auth) == false do
```

Comment:
Seems to make sense to directly add it to the pipeline instead of to each controller :) Not sure why it wasn't done in the first place 👍